src/Functions/FunctionsHashing.h

```diff
@@ -1240,17 +1310,31 @@ class FunctionAnyHash : public IFunction
 
         typename ColumnVector<ToType>::Container & vec_to = col_to->getData();
 
-        if (arguments.empty())
+        /// If using a "keyed" algorithm, the first argument is the key and
+        /// the data starts from the second argument.
+        /// Otherwise there is no key and all arguments are interpreted as data.
+        constexpr size_t first_data_argument = Keyed;
+
+        if (arguments.size() <= first_data_argument)
```

Comment:
(It has been like that before but the comment in l. 1320 is confusing ... it should just say that we produce a fixed, random-looking magic value if the input is empty)